RFS/src/main/java/org/opensearch/migrations/bulkload/transformers/TransformFunctions.java

```diff
@@ -78,17 +78,24 @@ public static void removeIntermediateMappingsLevels(ObjectNode root) {
 
     // Extract the mappings from their single-member list, will start like:
     // [{"_doc":{"properties":{"address":{"type":"text"}}}}]
+    //
+    // There should only be a single member in the list because multi-type mappings were deprecated in ES 6.X and
+    // removed in ES 7.X.  This list structure appears to be a holdover from previous versions of Elasticsearch.
+    // The exact name of the type can be arbitrarily chosen by the user; the default is _doc.  We need to extract
+    // the mappings from beneath this intermediate key regardless of what it is named.
     public static ObjectNode getMappingsFromBeneathIntermediate(ArrayNode mappingsRoot) {
+        if (mappingsRoot.size() != 1) {
+            throw new IllegalArgumentException("Mappings root does not contain a single member list");
+        }
         ObjectNode actualMappingsRoot = (ObjectNode) mappingsRoot.get(0);
-        if (actualMappingsRoot.has("_doc")) {
-            return (ObjectNode) actualMappingsRoot.get("_doc").deepCopy();
-        } else if (actualMappingsRoot.has("doc")) {
-            return (ObjectNode) actualMappingsRoot.get("doc").deepCopy();
-        } else if (actualMappingsRoot.has("audit_message")) {
-            return (ObjectNode) actualMappingsRoot.get("audit_message").deepCopy();
-        } else {
-            throw new IllegalArgumentException("Mappings root does not contain one of the expected keys");
+        ObjectNode actualMappingCopy = (ObjectNode) actualMappingsRoot.get(actualMappingsRoot.fieldNames().next()).deepCopy();
+
+        // Confirm the thing we retrieved has the shape we expect
+        if (!actualMappingCopy.has("properties")) {
+            throw new IllegalArgumentException("Mappings object does not contain the expected properties key");
```

Comment:
Okay, I'm not sure I understand exactly what's happening here and how it maps to the customer case.

My understanding is that their data structure looked like:
```
  {
    "mappings" : {
      "properties" : {
        "account_id" : ...,
        "next_field_here":...
      }
    }
  }
```

So `mappingsRoot` is the `mappings` item. This code seems to say that there needs to be a layer (`actualMappingsRoot`) between `mappings` and `properties`. But the customer JSON doesn't have a layer there. Am I missing understanding something here?

Reply:
There's two things going on here, 1. in the API, when `?include_type_names=false` which is the default, what you showed is returned In the API, when `?include_type_names=true`, then the type key is included, in the snapshot, this is always included (plus wrapping it in an array)

@chelma, IIRC, @peternied demo'd functionality to migrate metadata from a cluster instead of a snapshot. can you confirm that works with this?

Reply:
@AndreKurait Good call, lemme look into that.

Reply:
> @chelma, IIRC, @peternied demo'd functionality to migrate metadata from a cluster instead of a snapshot. can you confirm that works with this?

Great catch; this behavior is still broken.  Addressing it will also address Mikayla's concern about "what if there isn't an intermediary".